### PR TITLE
[ENDPOINT] POST /auth/register - OT102-32

### PR DESCRIPTION
**Pull request by gabodmc**
[OT102-32](https://alkemy-labs.atlassian.net/browse/OT102-32)

**AS :** user
**I WANT:** register my user
**TO :** become a user 

**Accepted criteria:** 
POST /auth/register must validate firstname, lastname, email and password fields. The password must be encrypted. Will return the new user

**Validate:** 
Encrypt password
Storage in DB

### Evidence
**Firstname validation:** 

**Lastname validation:** 

**Email validation:** 

**Email validation 2:** 

**Password validation 2:** 

**Created user:** 

**Email validated in db**

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -20,6 +21,27 @@ module.exports = {
...
+      const newUser = await addUser(userFormData)
+
+      endpointResponse({
+        res,
+        message: 'User were created successfully.',
+        body: newUser,
+        status: 201,
+      })
+    } catch (error) {
+      const httpError = createHttpError(500, `[Error retrieving users] - [users - get]: ${error.message}`)
```
Comment (dvelezroman) [inline controllers/users.js]: bien...

**Review by dvelezroman — APPROVED** [review]
Excellent

File: routes/index.js [hunk 2]
```diff
@@ -7,15 +7,18 @@ const pingRouter = require('./ping')
...
 const organizationsRouter = require('./organizations')
 const contactsRouter = require('./contacts')
 
 router.use('/users', usersRouter)
+router.use('/auth', authRouter)
 router.use('/ping', pingRouter)
 router.use('/categories', categoriesRouter)
 router.use('/members', membersRouter)
 router.use('/news', entriesRouter)
+router.use('/auth', authRouter)
```
Comment (JonatanGarbuyo) [inline routes/index.js]: te quedo'  doble?

**Review by gabyMangoni — APPROVED** [review]
All Ok

**Review by JonatanGarbuyo — APPROVED** [review]

**Pull request merged**